setup.sh

```diff
@@ -60,39 +60,74 @@ export REGION="us-central1"
 
-## Ops Project ##
-OPS_ENVIRONMENT_DIR=terraform/environments/ops
-export TF_VAR_project_id=${OPS_PROJECT}
-terraform -chdir=${OPS_ENVIRONMENT_DIR} init
-terraform -chdir=${OPS_ENVIRONMENT_DIR} apply --auto-approve
+        REPO_CONNECT_URL="https://console.cloud.google.com/cloud-build/triggers/connect?project=${OPS_PROJECT}"
+        echo "Connect your repos: ${REPO_CONNECT_URL}"
+        read -rp "Once your repo is connected, please continue by typing any key."
+        continue=1
+        while [[ ${continue} -gt 0 ]]; do
+            read -rp "Please input the repo owner [GoogleCloudPlatform]: " REPO_OWNER
+            repo_owner=${repo_owner:-GoogleCloudPlatform}
+            read -rp "Please input the repo name [emblem]: " REPO_NAME
+            repo_name=${repo_name:-emblem}
 
-## Staging Project ##
+            read -rp "Is this the correct repo: ${REPO_OWNER}/${REPO_NAME}? (y/n) " yesno
 
-STAGE_ENVIRONMENT_DIR=terraform/environments/staging
-export TF_VAR_project_id=${STAGE_PROJECT}
-export TF_VAR_ops_project_id=${OPS_PROJECT}
-terraform -chdir=${STAGE_ENVIRONMENT_DIR} init
-terraform -chdir=${STAGE_ENVIRONMENT_DIR} apply --auto-approve
+            if [[ ${yesno} == "y" ]]; then
+                continue=0
+            fi
+        done
 
-## Prod Project ##
-# Only deploy to separate project for multi-project setups
-if [ "${PROD_PROJECT}" != "${STAGE_PROJECT}" ]; then 
-PROD_ENVIRONMENT_DIR=terraform/environments/prod
-export TF_VAR_project_id=${PROD_PROJECT}
-export TF_VAR_ops_project_id=${OPS_PROJECT}
-terraform -chdir=${PROD_ENVIRONMENT_DIR} init
-terraform -chdir=${PROD_ENVIRONMENT_DIR} apply --auto-approve
-fi
+        # Configure terraform to setup the CD system.
+        export TF_VAR_cd_system="true"
+        export TF_VAR_repo_owner="${repo_owner}"
+        export TF_VAR_repo_name="${repo_name}"
+    fi # skip triggers
+
+    ## Terraform ##
+
+    echo
+    echo "$(tput bold)Setting up your Cloud resources with Terraform...$(tput sgr0)"
+    echo
+
+    # Ops Project
+    OPS_ENVIRONMENT_DIR=terraform/environments/ops
+    export TF_VAR_project_id=${OPS_PROJECT}
+    export TF_VAR_repo_owner=${OPS_PROJECT}
+    export TF_VAR_repo_name=${OPS_PROJECT}
+    terraform -chdir=${OPS_ENVIRONMENT_DIR} init
+    terraform -chdir=${OPS_ENVIRONMENT_DIR} apply --auto-approve
+
+    # Staging Project
+    STAGE_ENVIRONMENT_DIR=terraform/environments/staging
+    export TF_VAR_project_id=${STAGE_PROJECT}
+    export TF_VAR_ops_project_id=${OPS_PROJECT}
+    terraform -chdir=${STAGE_ENVIRONMENT_DIR} init
+    terraform -chdir=${STAGE_ENVIRONMENT_DIR} apply --auto-approve
+
+    # Prod Project
+    # Only deploy to separate project for multi-project setups
+    if [ "${PROD_PROJECT}" != "${STAGE_PROJECT}" ]; then 
+        PROD_ENVIRONMENT_DIR=terraform/environments/prod
+        export TF_VAR_project_id=${PROD_PROJECT}
+        export TF_VAR_ops_project_id=${OPS_PROJECT}
+        export TF_VAR_deploy_trigger_project=${STAGE_PROJECT}
```

Comment:
I assume `STAGE_PROJECT` (vs. `PROD_PROJECT`) isn't a typo here. 🙂

Reply:
No, it's not a typo. (Also, it will be gone in the next push.)

Reply:
fixed